ibis/expr/selectors.py

```diff
@@ -284,3 +307,137 @@ def matches(regex: str | re.Pattern) -> Selector:
     """
     pattern = re.compile(regex)
     return where(lambda col: pattern.search(col.get_name()) is not None)
+
+
+@public
+def any_of(*predicates: Predicate) -> Predicate:
+    """Include columns satisfying any of `predicates`."""
+    return functools.reduce(operator.or_, predicates)
+
+
+@public
+def all_of(*predicates: Predicate) -> Predicate:
+    """Include columns satisfying all of `predicates`."""
+    return functools.reduce(operator.and_, predicates)
+
+
+@public
+def c(*names: str) -> Predicate:
+    """Select specific column names."""
+    return where(lambda col, names=frozenset(names): col.get_name() in names)
+
+
+class Across(Selector):
+    __slots__ = "selector", "funcs", "names"
+
+    def __init__(
+        self,
+        selector: Selector,
+        funcs: Deferred
+        | Callable[[ir.Value], ir.Value]
+        | Mapping[str, Deferred | Callable[[ir.Value], ir.Value]],
+        names: Callable[[str, str], str] | None = None,
+    ) -> None:
+        self.selector = selector
+        self.funcs = funcs if isinstance(funcs, Mapping) else {None: funcs}
+        self.names = (
+            (lambda col, fn: col if fn is None else f"{col}_{fn}")
+            if names is None
+            else names
+        )
+
+    def expand(self, table: ir.Table) -> Sequence[ir.Value]:
+        expanded = []
+
+        names = self.names
+        cols = self.selector.expand(table)
+        for func_name, func in self.funcs.items():
+            resolve = func.resolve if isinstance(func, Deferred) else func
+            expanded.extend(
+                col.name(names(orig_col.get_name(), func_name))
+                for col, orig_col in zip(map(resolve, cols), cols)
+            )
+
+        return expanded
+
+
+@public
+def across(selector, func) -> Across:
+    return Across(selector, func)
+
+
+class IfAnyAll(Selector):
+    __slots__ = "selector", "predicate", "summarizer"
```

Comment:
You could use `Annotable` or `Concrete` here to reduce boilerplate.

Reply:
Done.